python_modules/dagster/dagster/_core/execution/plan/inputs.py

```diff
@@ -786,6 +786,46 @@ def _load_input_with_input_manager(
     from dagster._core.execution.context.system import StepExecutionContext
 
     step_context = cast(StepExecutionContext, context.step_context)
+
+    if context.has_asset_key:
+        # check that the asset has been materialized
+        instance = (
+            context.instance
+        )  # if use instance here need to test what happens when you make a context for unit testing
+        latest_input_materialization = instance.get_latest_materialization_event(
+            asset_key=context.asset_key
+        )
+        if latest_input_materialization is None:
```

Comment:
this check might cause user's tests to fail since we are requiring stricter enforcement that upstream assets be materialized. you can see test failures in our own test suite where we materialize an asset without materializing the upstreams

Reply:
additionally, this check makes self dependent assets not work since the base-case materialization fails at this condition. so need to figure that out as well